homeassistant/components/evohome/__init__.py

```diff
@@ -248,13 +171,26 @@ async def load_auth_tokens(store) -> tuple[dict, dict | None]:
         session=async_get_clientsession(hass),
     )
 
+    coordinator: DataUpdateCoordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=DOMAIN,
+        update_interval=config[DOMAIN][CONF_SCAN_INTERVAL],
+    )
+
     hass.data[DOMAIN] = {}
     hass.data[DOMAIN]["broker"] = broker = EvoBroker(
-        hass, client_v2, client_v1, store, config[DOMAIN]
+        hass, client_v2, client_v1, store, coordinator, config[DOMAIN]
     )
 
-    await broker.save_auth_tokens()
-    await broker.async_update()  # get initial state
+    coordinator.update_method = broker.async_update
```

Comment:
This is a bit of a code smell to me – `EvoBroker` contains the coordinator (to be able to update listeners), but the coordinator needs to be aware of the broker's update method.

I wonder if it would make sense to have `EvoBroker` extend `DataUpdateCoordinator`.

Reply:
>This is a bit of a code smell to me – EvoBroker contains the coordinator (to be able to update listeners), but the coordinator needs to be aware of the broker's update method.

Isn't this how the entities are set up in the absence of a broker?

For example, there is a similar setup between `asyncio.Protocol` and `asyncio.Transport`:
 - you instantiate the transport, and then
 - pass the transport as an arg when instantiating the protocol, then finally
 - call transport.set_protocol(protocol)

>I wonder if it would make sense to have `EvoBroker` extend `DataUpdateCoordinator`.

I had thought about it... But I'd like keep to keep it the way it is:
 - they're nicely 'loosely coupled': `broker.async_update` & `cordinator.async_update_listeners` only
 - I wouldn't be extending any coordinator methods, except possibly `coordinator._async_update_data` & isn't that what the `coordinator.update_method` instance attr is for?
 - there is no redundant code between the two